ext/standard/proc_open.c

```diff
@@ -36,6 +36,10 @@
 #include <fcntl.h>
 #endif
 
+#ifdef PHP_WIN32
+# pragma warning(disable: 4024 4047)
+#endif
```

Comment:
Can you share which warnings are produced here?

Reply:
See https://ci.appveyor.com/project/php/php-src/builds/33115213/job/ahmlcbah7a84hh5s?fullLog=true#L1225 ff. The problem is that `php_file_descriptor_t` is defined as `HANDLE`, but may actually be an fd (`int`). I think @alexdowad can give some insights.

Reply:
I don't have time to look at all of them right now, but I can see that one of the warnings is for a function which is not used on Win32 but which I didn't wrap in `#ifndef PHP_WIN32..#endif`. (That's `close_parentends_of_pipes`.)

My thought was that if the compiler sees all the code (or as much of it as possible) on all platforms, it means compiler syntax checking, etc. can catch as many issues as possible before the code goes to CI. But now I see this is causing unwanted compiler warnings.

I suggest you wrap `close_parentends_of_pipes` in `#ifndef PHP_WIN32`. As mentioned, I'll look at the others fairly soon.

Reply:
Thanks! Not defining `close_parentends_of_pipes()` on Windows already the warnings proc_open.c. No need to look further. :)

Reply:
Looks like `_php_array_to_envp` is wrong to declare `cnt` as `size_t`. That should have been `uint32_t`.

Reply:
Should I just open a PR to kill the warnings on `proc_open.c`?

Reply:
You mean other warnings? Yes, a PR would be great!